### PR TITLE
[rewrite] Add a couple new flags to LineTrace, for convenience

### PR DESCRIPTION
**Pull request by OrdinaryMagician**
Second attempt for this, since I had to add support to the new BLOCKASPLAYER flag.

TRF_SOLIDACTORS : Makes the trace get blocked by solid actors, in addition to the default of just shootable ones (which can be disabled).
TRF_BLOCKUSE : I figured that since LineTrace can be easily used as part of direct line activation now that that one feature is in, it would need some way to behave like real player use and be blocked by lines with "block use".
TRF_BLOCKSELF : This flag makes the trace stop at lines that would normally block the tracing actor (e.g.: block projectiles for missile flagged actors).

Here's an example for all three flags: [linetrace_newflags_m.zip](https://github.com/coelckers/gzdoom/files/2072865/linetrace_newflags_m.zip)

One of the tests shows TRF_BLOCKUSE in action with a linetrace-based remote switch activator. The other shows a beam trace with TRF_SOLIDACTORS stop at an actor that is solid, but not shootable.

In a separate room there is a custom imp that uses LineTrace to launch itself towards the target, bouncing off any world geometry that blocks it, including the "block monsters" lines surrounding it, thanks to TRF_BLOCKSELF.

### CODE REVIEW

File: src/p_map.cpp [hunk 1]
```diff
@@ -4969,9 +4969,21 @@ bool P_LineTrace(AActor *t1, DAngle angle, double distance,
...
-	int lflags = 0;
-	if ( !(flags & TRF_THRUBLOCK) ) lflags |= ML_BLOCKEVERYTHING;
-	if ( !(flags & TRF_THRUHITSCAN) ) lflags |= ML_BLOCKHITSCAN;
+	if ( flags & TRF_SOLIDACTORS ) aflags |= MF_SOLID;
+	int lflags = ML_BLOCKEVERYTHING|ML_BLOCKHITSCAN;
+	if ( flags & TRF_BLOCKUSE ) lflags |= ML_BLOCKUSE;
+	if ( flags & TRF_BLOCKSELF )
+	{
+		bool Projectile = ((t1->flags&MF_MISSILE)||(t1->BounceFlags&BOUNCE_MBF));
+		bool NotBlocked = ((t1->flags3&MF3_NOBLOCKMONST)||((i_compatflags&COMPATF_NOBLOCKFRIENDS)&&(t1->flags&MF_FRIENDLY)));
```
[the remark this answers is not in the document]
Comment (OrdinaryMagician) [reply src/p_map.cpp]: Oh, didn't notice. I usually don't add spacing in "oneliner" assignments on my own code so I guess I slipped into it here too.

**Comment by OrdinaryMagician** [comment]
Any chance that this will make it in? Should I change up anything?

**Comment by OrdinaryMagician** [comment]
I added the requested spacing, what now?

**Comment by alexey-lysiuk** [comment]
Sorry, such things are up to Graf to decide.

**Pull request merged**